htdocs/product/stock/replenish.php

```diff
@@ -839,7 +839,9 @@
 		if ($usevirtualstock) {
 			// If option to increase/decrease is not on an object validation, virtual stock may differs from physical stock.
 			$stock = $prod->stock_theorique;
-			//TODO $stockwarehouse = $prod->stock_warehouse[$fk_entrepot]->;
+			if (!empty($conf->global->STOCK_ALLOW_VIRTUAL_STOCK_PER_WAREHOUSE)) { //if conf active, stock virtual by warehouse is calculated
```

Comment:
Can you replace the 
`!empty($conf->global->STOCK_ALLOW_VIRTUAL_STOCK_PER_WAREHOUSE)`
with 
`getDolGlobalString('STOCK_ALLOW_VIRTUAL_STOCK_PER_WAREHOUSE')`

$conf->global->... must not be used anymore.